ops/model.py

```diff
@@ -1785,6 +1786,19 @@ def __iter__(self):
     def __getitem__(self, key: Union['Unit', 'Application']) -> 'RelationDataContent':
         return self._data[key]
 
+    def __setitem__(self, key: Union['Unit', 'Application'], value: Dict[str, str]) -> None:
+        # v0 -- make a new RelationDataContent on assignment
+        content = RelationDataContent(relation=self.relation, entity=key, backend=self._backend)
+        content.update(value)
+        self._data[key] = content
+        return
+        # v1 -- replace contents of existing RelationDataContent object
+        if key not in self._data:
+            raise KeyError(key, 'Assignment only supported for known keys.')
```

Comment:
Same comment as Ben here: the keys are app/unit, so it seems like they would always be present, unless I'm missing something.

You would want to avoid things like `relation.data['foo'] = {...}` or relation.data[Unit('made-up-unit')] = {...}`, but I think you'd want the error message to be clearer than just "known keys".